app/models/pglogical_subscription.rb

```diff
@@ -28,6 +28,19 @@ def self.lookup_by_id(to_find)
     nil
   end
 
+  def self.with_connection_error_handling
+    retry_attempted = false
+    yield
+  rescue PG::ConnectionBad => bad_connection_err
+    if retry_attempted
+      raise bad_connection_err
+    else
+      pglogical(true)
+      retry_attempted = true
+      retry
+    end
```

Comment:
You can just say `raise` with no parameter to have it re-raise (which I think is preferable because if I recall, it produces a better backtrace)...on that note it might be cleaner to use a guard clause.

```suggestion
  rescue PG::ConnectionBad => bad_connection_err
    raise if retry_attempted

    pglogical(true)
    retry_attempted = true
    retry
```

Reply:
Just like below, I did test that this worked as expected.  Here is the script I used:

```ruby
def foo
  raise NoMethodError
end

def bar
  retry_attempted ||= false
  foo
rescue NoMethodError => e
  raise e if retry_attempted

  puts "retrying..."
  retry_attempted = true
  retry
end

bar

```

Worth noting:  Changing this had no effect

```diff
-rescue NoMethodError => e
-  raise e if retry_attempted
+rescue NoMethodError
+  raise if retry_attempted
```

And the same output was displayed for either version:

```
$ ruby reraise_with_e.rb
retrying...
Traceback (most recent call last):
        2: from reraise.rb:16:in `<main>'
        1: from reraise.rb:7:in `bar'
reraise.rb:2:in `foo': NoMethodError (NoMethodError)
$ ruby reraise_without_e.rb
retrying...
Traceback (most recent call last):
        2: from reraise.rb:16:in `<main>'
        1: from reraise.rb:7:in `bar'
reraise.rb:2:in `foo': NoMethodError (NoMethodError)
```

That said, I agree that it looks cleaner, so I will change.